From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 09 Dec 2016 22:50:13 +0000
Subject: SCSI: Fix ABI change in 3.16.37
Forwarded: not-needed

Commit f05795d3d771 ("scsi: Add intermediate STARGET_REMOVE state to
scsi_target_state") added an enumerator to enum scsi_target_state,
and renumbered the existing STARGET_DEL.

The target state doesn't appear to be checked anywhere outside of the
SCSI core, but just in case it is swap STARGET_REMOVE and
STARGET_DEL to maintain the value of the latter.

Hide the change from genksyms.

---
--- a/include/scsi/scsi_device.h
+++ b/include/scsi/scsi_device.h
@@ -264,8 +264,10 @@ struct scsi_dh_data {
 enum scsi_target_state {
 	STARGET_CREATED = 1,
 	STARGET_RUNNING,
-	STARGET_REMOVE,
 	STARGET_DEL,
+#ifndef __GENKSYMS__
+	STARGET_REMOVE,
+#endif
 };
 
 /*
